packages/core/src/lib/icon/icon.svelte

```diff
@@ -41,22 +41,41 @@ export let size: Size = 'base';
 /** Additional CSS classes to pass to the svg. */
 let extraClasses: cx.Argument = '';
 export { extraClasses as cx };
+
+let allPaths: CustomIcon[] = [];
+
+$: {
+  const pathValue = paths[name];
+
+  if (typeof pathValue === 'string') {
+    allPaths = [{ path: pathValue }];
+  } else if (Array.isArray(pathValue)) {
+    allPaths = pathValue.map((icon) => ({
+      path: icon.path,
+      opacity: icon.opacity,
+    }));
+  }
+}
 </script>
 
 <!--
   Accessibility approach for icon svgs taken from:
   https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label
 -->
+
 <svg
   class={cx(sizes[size], extraClasses)}
-  viewBox="0 0 24 24"
+  viewBox={name === 'viam-mixed-part-status' ? '0 0 16 16' : '0 0 24 24'}
```

Comment:
I still needed to add a ternary here to set the correct viewbox for the mixed part status icon.  Is there a way that I can edit the viewbox of the mixed part status svg myself to match that of the other icons?

Reply:
Scaled to 24px!

```svg
<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none">
  <path fill="currentColor" d="M12 11c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2Zm6 2c0-3.31-2.69-6-6-6s-6 2.69-6 6c0 2.22 1.21 4.15 3 5.19l1-1.74c-1.19-.7-2-1.97-2-3.45 0-2.21 1.79-4 4-4s4 1.79 4 4c0 1.48-.81 2.75-2 3.45l1 1.74c1.79-1.04 3-2.97 3-5.19Z"/>
  <path fill="currentColor" d="M12 3C6.48 3 2 7.48 2 13c0 3.7 2.01 6.92 4.99 8.65l1-1.73C5.61 18.53 4 15.96 4 13c0-4.42 3.58-8 8-8s8 3.58 8 8c0 2.96-1.61 5.53-4 6.92l1 1.73c2.99-1.73 5-4.95 5-8.65 0-5.52-4.48-10-10-10Z" opacity=".5"/>
</svg>
```

Steps:

1. Export SVG from Figma
2. Drop into a new Figma file
3. Set to scale from center
4. Increase size from 16x16 to 24x24
5. Re-export
6. Remove cruft from SVG manually
7. Run through svgomg.com for good measure

Reply:
Thank you MC!